File: datasets/java_source-code/linked_list/type_3_structural.java
public class Main {
    public static boolean linearSearch(int[] arr, int target) {
        int index = 0;
        boolean found = false;

        while (index < arr.length && !found) {
            if (arr[index] == target) {
                found = true;
            }
            index++;
        }

        return found;
    }

    public static void main(String[] args) {
        int[] nums = {3, 7, 1, 9, 5};
        System.out.println(linearSearch(nums, 9));
    }
}

File: server/datasets/python_source-code/linked_list/type_3_structural.py
def linear_search(arr, target):
    index = 0
    found = False
    while index < len(arr) and not found:
        if arr[index] == target:
            found = True
        index += 1
    return found

nums = [3, 7, 1, 9, 5]
print(linear_search(nums, 9))
